lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -0,0 +1,20 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class Configuration
+    extend T::Sig
+
+    sig { returns(T::Array[String]) }
+    attr_accessor :files_to_index
+
+    sig { void }
+    def initialize
+      files_to_index = $LOAD_PATH.flat_map { |p| Dir.glob("#{p}/**/*.rb", base: p) }
```

Comment:
What about `.ru` and `.rake` files? Should this be configurable? Maybe the constructor could take the list of extensions with a default:

```rb
    def initialize(allowed_extensions: [".rb"])
```

Reply:
Yes, the follow up PRs will add more configuration options that allow you to add other patterns.